src/AutoRest.CSharp.V3/ClientModels/ClientBuilder.cs

```diff
@@ -0,0 +1,255 @@
+                query.ToArray(),
+                headers.ToArray(),
+                httpResponse.StatusCodes.Select(ToStatusCode).ToArray(),
+                body
+            );
+
+            ClientTypeReference? responseType = null;
+            if (response is SchemaResponse schemaResponse)
+            {
+                var schema = schemaResponse.Schema is ConstantSchema constantSchema ? constantSchema.ValueType : schemaResponse.Schema;
+                responseType = ClientModelBuilderHelpers.CreateType(schema, isNullable: false);
+            }
+
+            return new ClientMethod(
+                operation.CSharpName(),
+                request,
+                methodParameters.ToArray(),
+                responseType
+            );
+        }
+
+        private static QuerySerializationStyle GetSerializationStyle(HttpParameter httpParameter, Schema valueSchema)
+        {
+            Debug.Assert(httpParameter.In == ParameterLocation.Query);
+
+            switch (httpParameter.Style)
+            {
+                case null:
+                case SerializationStyle.Form:
+                    return valueSchema is ArraySchema ? QuerySerializationStyle.CommaDelimited : QuerySerializationStyle.Simple;
+                case SerializationStyle.PipeDelimited:
+                    return QuerySerializationStyle.PipeDelimited;
+                case SerializationStyle.SpaceDelimited:
+                    return QuerySerializationStyle.SpaceDelimited;
+                case SerializationStyle.TabDelimited:
+                    return QuerySerializationStyle.TabDelimited;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private static ConstantOrParameter[] ToParts(string httpRequestUri, Dictionary<string, ConstantOrParameter> parameters)
+        {
+            List<ConstantOrParameter> host = new List<ConstantOrParameter>();
+            foreach ((string text, bool isLiteral) in GetPathParts(httpRequestUri))
+            {
+                host.Add(isLiteral ? ClientModelBuilderHelpers.StringConstant(text) : parameters[text]);
+            }
+
+            return host.ToArray();
+        }
+
+        private static PathSegment[] ToPathParts(string httpRequestUri, Dictionary<string, PathSegment> parameters)
+        {
+            PathSegment TextSegment(string text)
+            {
+                return new PathSegment(ClientModelBuilderHelpers.StringConstant(text), false, SerializationFormat.Default);
+            }
+
+            List<PathSegment> host = new List<PathSegment>();
+            foreach ((string text, bool isLiteral) in GetPathParts(httpRequestUri))
+            {
+                if (!isLiteral)
+                {
+
+                    if (!parameters.TryGetValue(text, out var segment))
+                    {
+                        //TODO: WORKAROUND FOR https://github.com/Azure/autorest.modelerfour/issues/58
```

Comment:
It looks like this was fixed now. We should probably try to resolve this workaround.